src/main/java/com/mael/ttt/Turn.java

```diff
@@ -20,31 +23,28 @@ public void placeMark(Player player) {
         board.setCell(player.getMove(board), player.getMark().getString());
     }
 
-    public boolean isNotGameOver() {
-        return !(hasWinner() || checkForFull());
-    }
-
-    private boolean hasWinner() {
-        return checkForWinner(Mark.PLAYER) || checkForWinner(Mark.OPPONENT);
+    public boolean canBePlayed() {
+        return !(hasWinner() || isFull());
```

Comment:
I had the same comment from @ecomba @me-stevens  :-) Now I have two methods - one for 'hasWinner' and one for 'hasNoWinnerYet' so the client code can choose which one to call.